job-dsls/jobs/FlowJob.groovy

```diff
@@ -405,8 +409,36 @@ mvn -B -e org.sonatype.plugins:nexus-staging-maven-plugin:1.6.5:deploy-staged-re
  -DrepositoryDirectory=$deployDir -s $SETTINGS_XML_FILE -DstagingProfileId=15c3321d12936e -DstagingDescription="kie $kieVersion" -DstagingProgressTimeoutMinutes=40
 # creates a file (list) of the last commit hash of each repository as handover for production
 ./droolsjbpm-build-bootstrap/script/git-all.sh log -1 --pretty=oneline >> git-commit-hashes.txt
-echo $kieVersion > $WORKSPACE/version.txt'''
-
+echo $kieVersion > $WORKSPACE/version.txt
+# creates JSON file for prod
+# resultant sed extraction files
+./droolsjbpm-build-bootstrap/script/git-all.sh log -1 --format=%H  >> sedExtraction_1.txt
+sed -e '1d;2d' -e '/Total/d' -e '/====/d' -e 's/Repository: //g' -e 's/^/\\"/; s/$/\\"/;' -e '/""/d' sedExtraction_1.txt >> sedExtraction_2.txt
+sed -e '$!N;s/\\n/ /g' -e 's/ / : /g' -e 's/$/ ,/' -e '$ s/.$//' sedExtraction_2.txt >> sedExtraction_3.txt
+cat sedExtraction_3.txt
+fileToWrite=$reportDate.json
+commitHash=$(cat sedExtraction_3.txt)
+cat <<EOF > int.json
+{
+   "handover" : {
+   "cut_off_date" : "$cutOffDate",
+   "report_date": "$reportDate",
+   "repos" : [
+      {
+         $commitHash
+      }
+    ],
+   "source_product_tag":"$sourceProductTag",
+   "target_product_build":"$targetProductBuild" 
+   }
+}
+EOF
+# indent json
+python -m json.tool int.json >> $fileToWrite
```

Comment:
Following is mostly my personal opinion, so feel free to ignore it. I am not sure this indenting justifies the need to install and maintain Python on the slaves. I agree that we need some way to script this, but I was more thinking of using groovy.

In any case, we will need to make sure that either groovy or python are installed on the machines which will run this script.

Reply:
@psiroky - some useful comments! are there any cases where Python would not be on the slaves? afaik, there are a lot of built in system tools that are written in Python and I was always under the impression for this reason, it would always be available. IIRC, dnf for a long time was mostly Python.

Either way, I guess this or Groovy is fine, but being conscious of reducing maintenance, I think using built in Python libraries are probably OK. 

WDYT?

Reply:
TBH, I haven't checked, so it may very well be already there (coincidentally with something else). So if that's the case, then this is probably not an issue.

I do not like Python at all, but I'll not let this hold this PR and the other possible uses :)

Reply:
My apologies. The slaves do have `Python 2.7.5` installed as of now.